File: solutions/python/VB_T5_003_safe_division_exceptions.py
"""VB-T5-003: Safe Division Exceptions -- Python baseline."""
def safe_div(a: int, b: int) -> int:
    try:
        if b == 0: raise ValueError(-1)
        return a // b
    except ValueError as e:
        return e.args[0]

if __name__ == "__main__":
    assert safe_div(10, 2) == 5
    assert safe_div(7, 0) == -1
    print("VB-T5-003 ok")

File: solutions/typescript/VB_T5_003_safe_division_exceptions.ts
// VB-T5-003: Safe Division -- TypeScript baseline
function safeDiv(a: number, b: number): number {
  if (b === 0) return -1;
  return Math.trunc(a / b);
}
console.assert(safeDiv(10, 2) === 5);
console.assert(safeDiv(7, 0) === -1);
